wps_tools/utils.py

```diff
@@ -170,3 +171,23 @@ def log_handler(
     log_file_path = Path(process.workdir) / log_file_name  # From Finch bird
     log_file_path.open("a", encoding="utf8").write(message + "\n")
     response.update_status(message, status_percentage=status_percentage)
+
+
+def copy_http_content(http):
+    """
+        This function is implemented to make a temporary copy of files 
+        provided in http address that are inaccessible to the content
+        without downloading.
+
+        Parameters:
+            http (str): http address
+        Returs:
+            Path to the copied file in /tmp directory
+        """
+    http_content = get(http).content
+
+    tmp_copy = NamedTemporaryFile(
+        suffix=".nc", prefix="tmp_copy", dir="/tmp", delete=False,
+    )
+    tmp_copy.write(http_content)
+    return tmp_copy.name
```

Comment:
This file does create a temporary file, but because of the way we are using it here it does not get removed. Having `delete=False` prevents the temp file from being removed. On top of that returning a temp file is tricky, how do we ensure it gets removed when we are done with it? The best way I can think of is to bake this method into a custom context manager. At that point though I feel we may be over-engineering the solution, we could simply write the results to a file that is passed in. I favour the latter option, but either way this code will have to change.